### PR TITLE
ci(speed): don't run all tests to check PoE

### PR DESCRIPTION
**Pull request by pauldambra**
we run all the tests twice, once with PoE on and once with it off

but we only need to test PoE on tests that query insights.

Those aren't totally straight-forward to identify so I run the tests in some whole directories

docker compose up takes a pretty consistent 1 minute, which is probably as good as it gets, but we can skip it in the code quality checks and save 1 minute at least

## results

after a very scientific checking of two runs this saves about 3 minutes per run of the backend CI workflow

### CODE REVIEW

File: posthog/queries/time_to_see_data/serializers.py [hunk 1]
```diff
@@ -35,6 +34,8 @@ class SessionResponseSerializer(serializers.Serializer):
     user = serializers.SerializerMethodField()
 
     def get_user(self, session):
+        from posthog.api.shared import UserBasicSerializer
```
Comment (pauldambra) [inline posthog/queries/time_to_see_data/serializers.py:37]: to avoid a circular import

File: .github/actions/run-backend-tests/action.yml [hunk 2]
```diff
@@ -91,18 +91,25 @@ runs:
 
         # Tests
 
-        - name: Run FOSS tests
-          if: ${{ inputs.segment == 'FOSS' }}
-          env:
-              PERSON_ON_EVENTS_V2_ENABLED: ${{ inputs.person-on-events }}
-              GROUPS_ON_EVENTS_ENABLED: ${{ inputs.person-on-events }}
```
Comment (Twixes) [inline .github/actions/run-backend-tests/action.yml:98]: Where did these go?

File: .github/actions/run-backend-tests/action.yml [hunk 3]
```diff
@@ -91,18 +91,25 @@ runs:
...
               pytest hogvm posthog -m "not async_migrations" \
                   --splits ${{ inputs.concurrency }} --group ${{ inputs.group }} \
                   --durations=100 --durations-min=1.0 --store-durations \
                   $PYTEST_ARGS
 
+        - name: Run FOSS tests when person-on-events is true
+          if: ${{ inputs.segment == 'FOSS' &&  inputs.person-on-events == 'true' }}
+          shell: bash
+          run: | # async_migrations are covered in ci-async-migrations.yml
+              pytest ./posthog/clickhouse ./posthog/queries ./posthog/api/test/test_insight* \
```
Comment (Twixes) [inline .github/actions/run-backend-tests/action.yml:107]: I think it's also worth adding `./posthog/hogql`

**Review by Twixes — COMMENTED** [review]
I was thinking of using a `persons_querying` marker… but this is just going to be quicker to get in. Just left a couple of comments

File: .github/actions/run-backend-tests/action.yml (hunk 2 above)
Comment (pauldambra) [reply .github/actions/run-backend-tests/action.yml:98]: 🙈

**Review by Twixes — APPROVED** [review]
Also narrowed down EE tests - to avoid both steps being completely doubled, I put the paths of tests under a ternary conditional, so this should be a bit more concise. Let me know if this looks good :)

**Comment by pauldambra** [comment]
there's something messy here that I'm not sure of... quick to recreate the PR

**Pull request closed without merging**